File: src/types/Log.ts
export type Log = {
    id: string;
    habitId: string;
    logDate: Date;
};

export type LogDTO = {
    id: string;
    habit_id: string;
    log_date: string;
}

export type CreateLogPayload = {
    habitId: string;
    logDate: Date;
}

export type CreateLogInput = {
    habit_id: string;
    log_date: string;
}

export type DeleteLogPayload = {
    habitId: string;
    logId: string;
}

export type DeleteLogInput = {
    habit_id: string;
    log_id: string;
}

File: src-tauri/src/models/log.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateLogInput {
    pub habit_id: String,
    pub log_date: String,
}

#[derive(Debug, Deserialize)]
pub struct DeleteLogInput {
    pub habit_id: String,
    pub log_id: String,
}

#[derive(Debug, Serialize)]
pub struct Log {
    pub id: String,
    pub habit_id: String,
    pub log_date: String,
    pub created_at: String,
    pub updated_at: Option<String>,
}
